src/engine/renderer/glsl_source/computeLight_fp.glsl

```diff
@@ -75,9 +75,21 @@ void ReadLightGrid(in vec4 texel, out vec3 ambientColor, out vec3 lightColor) {
 }
 #endif
 
-#if defined(USE_DELUXE_MAPPING) || defined(USE_GRID_DELUXE_MAPPING) || (defined(r_realtimeLighting) && r_realtimeLightingRenderer == 1)
+#if defined(USE_DELUXE_MAPPING) || defined(USE_GRID_DELUXE_MAPPING) || defined(r_realtimeLighting)
+#if !defined(USE_PHYSICAL_MAPPING)
+#if defined(r_specularMapping)
 uniform vec2 u_SpecularExponent;
 
+vec3 computeSpecularity(vec3 lightColor, vec4 materialColor, float NdotH)
+{
+	return lightColor * materialColor.rgb * pow(NdotH, u_SpecularExponent.x * materialColor.a + u_SpecularExponent.y) * r_SpecularScale;
```

Comment:
This refactor makes sure the code using `u_SpecularExponent` is right next to the declaration of it, within the same `ifdef`, so it reduces the risk to reintroduce the bug in the future.